### PR TITLE
[WIP][SPARK-15953][SQL][STREAMING] Renamed ContinuousQuery to StreamingQuery

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?

Renamed for simplicity, so that its obvious that its related to streaming. 
The PR is blocked on #13653 and contains its changes as well. Hence its marked WIP.
## How was this patch tested?

Existing unit tests.

### CODE REVIEW

**Comment by tdas** [comment]
@zsxwing @marmbrus Could you take a look?

**Comment by marmbrus** [comment]
LGTM

File: python/pyspark/sql/readwriter.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 from pyspark.sql.types import *
 from pyspark.sql import utils
 
-__all__ = ["DataFrameReader", "DataFrameWriter"]
+__all__ = ["DataFrameReader", "DataFrameWriter", "DataStreamReader", "DataStreamWriter"]
```
Comment (tdas) [inline python/pyspark/sql/readwriter.py]: Based on comment in #13653

**Comment by tdas** [comment]
I am merging this patch to reduce conflicts. Merging to master and 2.0

**Comment by zsxwing** [comment]
LGTM.

**Comment by zsxwing** [comment]
By the way, I noticed that we don't have unit tests for Python DataStreamReader and DataStreamWriter. That can be done in a separate PR.

**Comment by zsxwing** [comment]
Thanks! Merging to master and 2.0.

**Pull request closed without merging**